src/Dropdown.js

```diff
@@ -141,8 +139,9 @@ class Dropdown extends React.Component {
     const container = this.getContainer();
     const menu = this.getMenu();
     const clickIsInContainer = container.contains(e.target) && container !== e.target;
+    const clickIsInInput = container.classList.contains('input-group') && container.classList.contains('dropdown') && e.target.tagName === 'INPUT';
     const clickIsInMenu = menu && menu.contains(e.target) && menu !== e.target;
-    if ((clickIsInContainer || clickIsInMenu) && (e.type !== 'keyup' || e.which === keyCodes.tab)) {
+    if (((clickIsInContainer && !clickIsInInput) || clickIsInMenu) && (e.type !== 'keyup' || e.which === keyCodes.tab)) {
```

Comment:
The removal of the `input-group-addon` class/component interfered with the ability for a a dropdown attached to an input group to automatically close when the input is clicked. This is because we no longer have the input group addon as a container. Adding a placeholder container interferes with the way Bootstrap 5 calculates button radius for buttons at the start/end of the input group, so this is a workaround that addon buttons are still rendered correctly while behavior of dropdowns within in input group remain correct.